libvirt: fix including libpcap and add xen support

globin opened the pull request:
see #4684, this also adds xen support to libvirt.

globin committed:
libvirt: fix including libpcap

globin committed:
libvirt: add xen support

globin committed:
libvirt-glib: 0.2.0 -> 0.2.3

This also fixes the build with the xen-supporting libvirt and removes
obsolete workarounds.

globin committed:
virt-viewer: fix build and clean up

domenkozar commented:
```
checking for promoted mode_t type... libvirt-glib.c:16:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
```

globin committed:
libosinfo: 0.2.11 -> 0.2.12

Unbreaks gnome-boxes 3.18

globin committed:
gnome-boxes, libguestfs: fix builds due to libvirt

Adds link time dependencies which are now used in libvirt and unbreaks
these packages

globin commented:
I'm sorry I can't reproduce that one, found more broken builds though after hours of building stuff, some definitely broken for some time, only thing now failing for me is pypy itself and pypyPackages.libvirt as a downstream build from that but that is definitely unrelated.

globin commented:
```
robin@graffel [11:15:50] [~/dev/nixpkgs] [libvirt-pcap]
-> % nix-build -A libvirt-glib -A perlPackages.SysVirt -A python35Packages.libvirt -A collectd -A libosinfo -A libguestfs -A gnome3_18.gnome-boxes -A python27Packages.libvirt -A python34Packages.libvirt -A libvirt -A gnome3.gnome-boxes -A virtmanager -A virtviewer -A virtinst
/nix/store/cdvha4yvgfwp3sf2y4ni9kmsy06gff4k-libvirt-glib-0.2.3
/nix/store/vl4cdhvr2fk4mkfwvwf63w2503s1mc3y-perl-Sys-Virt-1.2.19
/nix/store/31d0yckjgrxg3i0285y1xqb1ynbij6g6-libvirt-python-1.2.19
/nix/store/l6y4zff8briys3bjxwl6pf3814k46x8f-collectd-5.5.0
/nix/store/vkvjkqflvdlf2jqlq58jrd3zj6v5vbc0-libosinfo-0.2.12
/nix/store/wk66i1yyddfz3kgkhn2pav3i9g8zkbbm-libguestfs-1.29.5
/nix/store/90lp92i5d899vcjh3fxlfmps1brs8xbb-gnome-boxes-3.18.0
/nix/store/sa13yjqy9dbrlhkgmbvsmkrv1bn37j4v-libvirt-python-1.2.19
/nix/store/zfibsaj48aq29l9y6kfdajqrm122pchl-libvirt-python-1.2.19
/nix/store/x3w4ad3mhfh8wyr2snxvzjj82m864rxj-libvirt-1.2.19
/nix/store/ic0wr1fjv4x35z2h58n9hpsyjysqj40s-gnome-boxes-3.16.1
/nix/store/wyryh10ii12r9s1v0s6q3jx4lpf06nk6-virt-manager-1.3.1
/nix/store/z84pjmglhhvlzx3v4ilw171gby8zrvzs-virt-viewer-2.0
/nix/store/vq47w7i15la2qy8i65zap98s4m2fvgfx-virtinst-0.600.4
```

globin commented:
rebased to solve conflicts

copumpkin commented on pkgs/development/libraries/libvirt/default.nix:
```diff
@@ -2,7 +2,7 @@
 , libtasn1, ebtables, libgcrypt, yajl, makeWrapper, pmutils, libcap_ng
 , dnsmasq, libnl, libpcap, libxslt, xhtml1, numad, numactl, perlPackages
-, curl, libiconv, gmp
+, curl, libiconv, gmp, pythonPackages, xen
```
I don't think `pythonPackages` is necessary here anymore, right?

globin commented:
fixed

domenkozar commented:
@globin blah sorry, could you rebase?

globin commented:
just a sec

globin commented:
done

globin commented:
Cheers!
